### PR TITLE
Add first storage implementation

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This adds a generic key value storage trait and a default implementation
based on the sled crate. This way we can exchange the implementation if
required. The default storage does automatic de-/serialization, which
can be used to store any arbitrary time in which supports serde.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added generic key value based storage trait and default implementation based on sled
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
This is ready for review. PTAL @mrunalp 

Two main changes beside the additions:
- Moved `MyImage` and `MyService` into `Service` to only have one structure owning the storage
- Renamed `sock-path` to `run-path` which now will be used as root for the storage as well as the socket.

**Comment by saschagrunert** [comment]
Had to rebase to resolve the conflict.

**Comment by saschagrunert** [comment]
Had to rebase to resolve the conflict.

File: src/config.rs [hunk 1]
```diff
@@ -29,12 +29,14 @@ pub struct Config {
 
     #[get = "pub"]
     #[clap(
-        default_value("/var/run/cri/cri.sock"),
-        env("CRI_SOCK_PATH"),
-        long("sock-path")
+        default_value("/var/run/cri"),
+        env("CRI_RUN_PATH"),
+        long("run-path"),
```
Comment (mrunalp) [inline src/config.rs:34]: I feel like we should keep the storage and socket/listen path separate.

Comment (saschagrunert) [reply src/config.rs]: Sure, added the storage path as separate config value.

File: src/image_service/image_fs_info.rs [hunk 2]
```diff
@@ -1,10 +1,10 @@
 use crate::{
     criapi::{ImageFsInfoRequest, ImageFsInfoResponse},
-    image_service::MyImage,
+    service::Service,
```
Comment (mrunalp) [inline src/image_service/image_fs_info.rs:3]: I like the refactoring here but maybe we can pick a more descriptive name than `Service`?

Comment (saschagrunert) [reply src/image_service/image_fs_info.rs:3]: Hm, something with GRPC? GrpcServer?

Comment (saschagrunert) [reply src/image_service/image_fs_info.rs:3]: @haircommander any thoughts on a good name?

Comment (haircommander) [reply src/image_service/image_fs_info.rs:3]: I would say CRIService

Comment (saschagrunert) [reply src/image_service/image_fs_info.rs]: Renamed it to `CRIService`

**Comment by saschagrunert** [comment]
Ready for review again, PTAL @mrunalp

**Review by mrunalp — APPROVED** [review]

**Pull request merged**